x-pack/plugins/viz_editor/public/components/main/main.tsx

```diff
@@ -43,96 +51,129 @@ function reducer(state: State, action: Action): State {
       return {
         ...state,
         loading: false,
-        time: action.time,
+        dataSource: action.dataSource,
+        rawData: action.rawData,
       };
     case 'loadError':
       return {
         ...state,
         loading: false,
         errorMessage: action.message,
       };
+    case 'setQuery':
+      return {
+        ...state,
+        query: action.query,
+        rawData: action.rawData,
+      };
     default:
       throw new Error(`Unknown action ${(action as any).type}`);
   }
 }
 
-function fetchInitialState(kfetch: any, dispatch: Dispatch<Action>) {
-  return kfetch({
-    pathname: '/api/viz_editor/example',
-  })
-    .then((data: any) =>
-      dispatch({
-        type: 'loaded',
-        time: new Date(data.time),
-      })
-    )
-    .catch(({ message }: any) =>
-      dispatch({
-        type: 'loadError',
-        message,
-      })
-    );
+async function fetchInitialState(kfetch: any, dispatch: Dispatch<Action>) {
+  const dataSource = await ds.load({ kfetch });
```

Comment:
Regarding state management - should the data panel handle the fetching and current state of fields in the index pattern itself? By doing so a plugin could use another way of fetching this information by switching out the panel (e.g. TSVB, which doesn't rely on a single index pattern). If we add it here into the frame, it will be set for all editor we build in the future